src/book/02-system/03-tools/04-nugan.mdx

```diff
@@ -0,0 +1,214 @@
+
+![Segmentation mask and a black and white image where the background of the segmentation mask is black and the rest white.](images/cyclegan_attention.png 'Segmentation image and corresponding attention mask.')
+
+After we get a fake real image, we use the attention mask to add back in the original background.
+
+```py
+masked_fake_B = fake_B*attention + real_A*(1-attention)
+```
+
+`fake_B*attention` will give us the non-background pixels of the fake real image. `real_A*(1-attention)` will give us the background pixels of the original real NUbpr image. Adding these both together, we get the fake non-background elements and the real background elements in one image. This result is then used for training. This removes the consideration of the background from the training, and instead lets the generator focus its attention on the non-background elements.
+
+Since we do not have an attention mask for images in the B domain, we focus more on cycling on the original real A image. From a real B image, we generate a fake A image. We do not cycle from here. From a real A image, we cycle on the resulting fake B image, and then again on the resulting cycled fake A image. All the cycles from the real A image can use the attention mask from the corresponding segmentation mask for A.
+
+The repository contains a pretrained base model that will map semi-synthetic soccer field images to real soccer field images. This model can then be fine-tuned to a specific field by training from the base model with a dataset of real images of a specific field. When training is run, without specifying to continue an existing fine-tuning trained model, the pretrained base will be loaded.
+
+## Using NUgan
+
+### Prerequisites
+
+This code has been tested on Windows with Anaconda and Linux Mint.
+
+Clone the repository
+
+```bash
+git clone https://github.com/NUbots/NUgan
+```
+
+Move into the cloned repository
+
+```bash
+cd NUgan
+```
+
+You can use either CPU or NVIDIA GPU with latest drivers, [CUDA](https://developer.nvidia.com/cuda-downloads) and [cuDNN](https://developer.nvidia.com/cudnn).
+
+If you are using Windows, we recommend using [Anaconda](https://docs.anaconda.com/anaconda/install/windows/).
+
+To set up with pip run
+
+```sh
+sudo apt install python-pip
+pip3 install setuptools wheel
+pip3 install -r requirements.txt
+```
+
+To set up with Anaconda (MacOS) you can create a new environment with the given environment file
+
+```sh
+conda env create -f environment.yml
+```
+
+In Windows, use the batch file rather than the environment file.
+
+```sh
+"./scripts/conda_deps.bat"
+```
+
+On non-Windows OS you could also use the provided shell script to install dependencies in Anaconda.
```

Comment:
I feel like people will get to this having already run the previous steps most of the time, so might as well put the script first and the manual install after